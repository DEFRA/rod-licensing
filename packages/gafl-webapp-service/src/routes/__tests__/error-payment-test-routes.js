import errorPaymentTestRoutes from '../error-payment-test-routes.js'

describe('Error payment test route handler', () => {
  describe('error route', () => {
    it('has a return value with a method of GET and path of /buy/payment-error', async () => {
      expect(errorPaymentTestRoutes).toMatchSnapshot()
    })
  })

  describe('ERROR_TESTING handler', () => {
    const errorTesting = errorPaymentTestRoutes[0].handler

    it.each([['pre-payment'], ['post-payment']])('throws error', async origin => {
      const mockRequest = {
        query: { origin }
      }
      const mockH = {}

      await expect(errorTesting(mockRequest, mockH)).rejects.toThrow()
    })
  })
})
