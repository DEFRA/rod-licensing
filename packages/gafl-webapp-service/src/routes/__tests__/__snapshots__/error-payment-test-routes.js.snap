// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error payment test route handler error route has a return value with a method of GET and path of /buy/payment-error 1`] = `
Array [
  Object {
    "handler": [Function],
    "method": "GET",
    "path": "/buy/payment-error",
  },
]
`;
